### ISSUE TITLE
Selection is not working on server-driven datagrid.

### ISSUE THREAD

**Issue by krlf** [vmware/clarity#4504]
## Describe the bug
The issue has been already [reported](https://github.com/vmware/clarity/issues/4477) but rejected.

The matter is that `*ngFor` is [required](https://clarity.design/documentation/datagrid/server-driven) to be used for server-driven datagrid:

> One important thing to note is that since you don't have all the data available at once, you cannot use the smart iterator *clrDgItems: it would sort, filter and paginate a subset of the data that has already gone through all that. So all you need to do is to go back to a simple *ngFor, which we support. 

 It contradicts [this comment](https://github.com/vmware/clarity/issues/4477#issuecomment-616733019) and the [example](https://clarity.design/documentation/datagrid/selection) of the "selection" feature because `*clrDgItems` is proposed to be used in `clr-dg-row` instead of `ngFor`.

So when `*ngFor` is used in `clr-dg-row` and `clrDgSelected` is set in `clr-datagrid` then datagrid is not rendered and the error is logged:

`TypeError: Cannot read property 'findIndex' of undefined at Selection.isLocked (http://localhost:4200/vendor.js:95297:61) at ClrDatagridRow.get clrDgSelectable [as clrDgSelectable] (http://localhost:4200/vendor.js:95476:32) at ClrDatagridRow_ng_template_5_div_4_Template (http://localhost:4200/vendor.js:84357:154) at executeTemplate (http://localhost:4200/vendor.js:29693:9) at refreshView (http://localhost:4200/vendor.js:29540:13) at refreshDynamicEmbeddedViews (http://localhost:4200/vendor.js:30898:21) at refreshView (http://localhost:4200/vendor.js:29563:9) at refreshDynamicEmbeddedViews (http://localhost:4200/vendor.js:30898:21) at refreshView (http://localhost:4200/vendor.js:29563:9) at refreshDynamicEmbeddedViews (http://localhost:4200/vendor.js:30898:21)`

## How to reproduce

https://stackblitz.com/edit/angular-uaqrwd

No actions are required. Please, see console log.

## Expected behavior

`*ngFor` should work with `clrDgSelected` in datagrids.
Datagrid should be rendered. No console errors are expected.

## Versions

**App**

* Angular: [9.1]
* Node: [12.16.2]
* Clarity: [3.1]

**Device:**

* Type: [PC]
* OS: [Linux]
* Browser [Chrome]
* Version [81.0.4044.122]

## Additional notes

Need to note that `clrDgSelected` and `*ngFor` work fine on Clarity v2.

**Comment by krlf** [comment]
It could be related to this improvement https://github.com/vmware/clarity/pull/3749

```TypeScript
  public isLocked(item: T): boolean {
    /**
     * The check for selectionType will boost the performence by NOT searching
     * into the array when there is no need for that.
     */
    if (this.canItBeLocked()) {
      const ref = this._items.trackBy(this._items.all.findIndex(maybe => maybe === item), item);
      return this.lockedRefs.indexOf(ref) > -1;
    }

    return false;
  }
```

Just a suggestion: it is not clear why `this._items.all` is used without check if there are elements in the array.

**Comment by skipgeldens** [comment]
We're stuk on the same issue in our project. To show that is was working on 2.41 I created this stackblitz: https://stackblitz.com/edit/clarity-light-theme-v2-vcqwz2 with exactly the same datagrid code as @krlf. 

Also, we don't run into the issue when we directly route (using the browser) to the component containing the datagrid, only when we route from another page to the datagrid.

**Comment by gnomeontherun** [comment]
@bdryanovski Can you take a look at this one?

**Comment by bdryanovski** [comment]
This is fixed with #4626
